### DISCUSSION TITLE
[Proposal]: return if true

### DISCUSSION THREAD

**Discussion by LaraSQP** [dotnet/csharplang#6787] [Language Ideas] [answered]
Millions of developers read and write code like this every day:

```
if( a == null )
  return;
```

I cannot be the first to suggest:

`return if (a == null);`

And in general

`return if (true);`

If there is interest, I can elaborate. Otherwise, please tell me so and I will close the issue.

**Comment by HaloFour** [comment, marked as answer]
See: #2966

**Reply by ricardoboss** [reply]
As I understand it, this issue talks about a shortcut to write

```
if (a)
	return x;
else
	return y;
```

but what the author proposed (and what I wanted to propose before I found this) is something like this:

```
return x if a;
```

Which only returns when `a` evaluates to `true` but is a `NOP` otherwise.

This should be another issue, shouldn't it?

**Reply by HaloFour** [reply]
There's also #2195 which explicitly asks for the Ruby-like syntax.

In C# you can already write that as:

```cs
if (a) return x;
```
